packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -27,9 +30,120 @@ def _ensure_routable():
     return _ensure_routable
 
 
+def vip_test(dcos_api_session, vip, vipaddr, samehost, vipnet, proxynet):
+    teststr = 'vip={} vipaddr={} samehost={} vipnet={} proxynet={}'.format(vip, vipaddr, samehost, vipnet,
+                                                                           proxynet)
+    logging.info('VIP_TEST START {}'.format(teststr))
+    agents = list(dcos_api_session.slaves)
+    random.shuffle(agents)
+    host1 = agents[0]
+    host2 = agents[0]
+    if not samehost:
+        host2 = agents[1]
+    logging.info('host1 is is: {}'.format(host1))
+    logging.info('host2 is is: {}'.format(host2))
+    origin_app, app_uuid = get_test_app_in_docker()
+    origin_app['container']['docker']['portMappings'][0]['labels'] = {'VIP_0': vip}
+    origin_app['container']['docker']['network'] = vipnet
+    if vipnet == 'USER':
+        origin_app['ipAddress'] = {'networkName': 'dcos'}
+    if vipnet == 'HOST':
+        origin_app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '
+        origin_app['cmd'] += '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py '
+        origin_app['cmd'] += '$PORT0'
+        origin_app['container']['docker']['portMappings'][0]['hostPort'] = 0
+        origin_app['container']['docker']['portMappings'][0]['containerPort'] = 0
+        origin_app['container']['docker']['portMappings'][0]['labels'] = {}
+        origin_app['portDefinitions'] = [{'labels': {'VIP_0': vip}}]
+
+    origin_app['constraints'] = [['hostname', 'CLUSTER', host1]]
+    origin_app['constraints'] = [['hostname', 'CLUSTER', host1]]
+
+    proxy_app = get_test_app_in_docker()[0]
+    proxy_app['container']['docker']['network'] = proxynet
+    proxy_app['constraints'] = [['hostname', 'CLUSTER', host2]]
+    if proxynet == 'USER':
+        proxy_app['ipAddress'] = {'networkName': 'dcos'}
+    if proxynet == 'HOST':
+        proxy_app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '
+        proxy_app['cmd'] += '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py '
+        proxy_app['cmd'] += '$PORT0'
+        proxy_app['container']['docker']['portMappings'][0]['hostPort'] = 0
+        proxy_app['container']['docker']['portMappings'][0]['containerPort'] = 0
+
+    returned_uuid = None
+
+    with contextlib.ExitStack() as stack:
+        stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(origin_app))
+        sp = stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(proxy_app))
+        cmd = '/opt/mesosphere/bin/curl -s -f -m 5 http://{}/test_uuid'.format(vipaddr)
+        returned_uuid = ensure_routable(cmd, sp)()
+        logging.info('returned_uuid is: {}'.format(returned_uuid))
+    assert(returned_uuid is not None)
+    assert(returned_uuid['test_uuid'] == app_uuid)
+    logging.info('VIP_TEST PASSED {}'.format(teststr))
+    return app_uuid
+
+
+@pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
+def test_vip(dcos_api_session):
```

Comment:
This should see some overhaul.

1) Keep `addrs` and one of `vipnets` and `proxynets` (call it something common).
2) Have only two external for-loops to track. Kill the outer for-loop to iterate over samehosts  [True, False]. See if this can folded inside the code.
3) call the variable names failed, passed, skipped something with a more meaningful identifer (easy).
4) See if using of `num` can be avoided. (Encode the ports within).

5) Ideally, if this is constructing the configurations of the ip-addresses, just construct them early and separately as a structure by itself into a list and then iterate over it to add to the threads. ?